flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -624,7 +635,12 @@ public List<String> listFunctions(String dbName) throws CatalogException {
 
   @Override
   public CatalogFunction getFunction(ObjectPath functionPath) throws FunctionNotExistException, CatalogException {
-    throw new FunctionNotExistException(getName(), functionPath);
+    CatalogFunction catalogFunction = partitionFunctions.get(functionPath.getObjectName());
```

Comment:
I see that you've removed the `listFunctions` method, but that doesn't actually solve the problem -- it introduces a new one. The problem is that these functions should not be defined in every database, but that's what this `getFunction` implementation does because it ignores `functionPath.getDatabaseName()`.

Whatever this method does, `listFunctions` should be consistent with its behavior. If this will load a function, then `listFunctions` should show it.

Reply:
My original idea was that these functions belong to the catalog (it may not appropriate now) and that the functions are registered when the catalog is created so that users can use them directly. So I provide the function in `getFunctions` and show the function in `listFunctions`. If the function is not displayed to the user now, and these functions cannot be get directly (also functions will not be directly registered in the catalog), should we mention these functions in the documentation, and provide an example for register and use function ?